File: client/src/api/publicChat.ts
import { api } from './axios';

export interface PublicChatResponse {
  reply: string;
  tokensUsed: number;
}

export const publicChatApi = {
  message: (text: string) =>
    api
      .post<{ data: PublicChatResponse }>('/public/chat/message', { text })
      .then((r) => r.data.data),
};

File: Android/client/lib/api/publicChat.dart
// ignore_for_file: file_names

import 'api_client.dart';

class PublicChatApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> message(Map<String, dynamic> data) =>
      _api.post('/public/chat/message', data: data);
}
